### ISSUE TITLE
[BUG] Auto redeploy on hidden model fails due to permission

### ISSUE THREAD

**Issue by arjunkumargiri** [opensearch-project/ml-commons#2033]
**What is the bug?**
Auto redeploy model feature is used to automatically deploy existing models in case of process restart or when a new node is added to the cluster. Auto redeploy for hidden model fails with permission failure with error message:

```
Exception occurred when auto redeploying the model, model id is: HIDDEN_MODEL_2, exception is: User doesn't have privilege to perform this operation on this model, skipping current model auto redeploy and starting next model redeploy!
```

**How can one reproduce the bug?**
1. Create a hidden model
2. Set cluster settings flag `plugins.ml_commons.model_auto_redeploy.enable` to true
3. Restart opensearch process.

**What is the expected behavior?**
Hidden models gets automatically redeployed

**Comment by arjunkumargiri** [comment]
Proposal: During re-deploy flow check if current state of the model is already deployed. If model is deployed, then bypass super admin check in case of hidden model.

**Comment by ylwu-amzn** [comment]
> Proposal: During re-deploy flow check if current state of the model is already deployed. If model is deployed, then bypass super admin check in case of hidden model.

Thanks @arjunkumargiri . 
Discussed with @rbhavna , this sounds good. So the logic will be: is super admin deploy model, that means this model should always stay as deployed and anyone can deploy it.  That seems risky to allow anyone to deploy. How about only allow cluster admin user to deploy ? That logic will be if super admin deployed the model, then the cluster admin can have permission to maintain the model status as deployed, other user still can't do it

**Comment by arjunkumargiri** [comment]
Providing permission only to cluster admin to redeploy seems valid. However, we should ensure that deploy request on a hidden model by users that does not have permission results in deploy_failed model state.

**Comment by dhrubo-os** [comment]
I just talked with @rbhavna. I think we can solve this problem in different way. We need to identify if the deploy request is coming from the code or from person. If the deploy request is coming from code (in this auto redeploy is code initiated) then deploy request will go through for hidden model otherwise block. 


I needed to tackle the similar problem for get model request. Example PR: https://github.com/opensearch-project/ml-commons/pull/1648/files

**Comment by arjunkumargiri** [comment]
Based on the PR,  is `isDirectRequest` flag provided by caller? if so what prevents any user from providing the flag?

**Comment by dhrubo-os** [comment]
> Based on the PR, is `isDirectRequest` flag provided by caller? if so what prevents any user from providing the flag?

In main branch, it updated to the [isUserInitiatedGetRequest](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/model/MLModelGetRequest.java#L37C13-L37C38)

This flag is not provided by the caller. When a GET REST request comes, we mark it as `true` [here](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/rest/RestMLGetModelAction.java#L59-L63)

Otherwise false if the GetRequest is generated by code: [ref](https://github.com/opensearch-project/ml-commons/blob/52626be603036ab862b16ba2810266ce5f3d2e6b/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java#L98)

**Comment by rbhavna** [comment]
Using the flag as mentioned by @dhrubo-os seems reasonable. I tried only providing access to admin to deploy when model is already in DELOYED state but on cluster restart, auto redeploy failed again with the access denied error. 

When I checked what is the user info when auto redeploy gets triggered, it seems the user is null, hence request failing for admin as well. So we just cannot restrict it to admin. And @dhrubo-os's logic is already implemented for predict API during internal transport call to get model. We can use something similar
